addons/account/models/account_move.py

```diff
@@ -1,349 +1,1874 @@
+    @api.model
+    def _get_default_invoice_incoterm(self):
+        ''' Get the default incoterm for invoice. '''
+        return self.env.company.incoterm_id
+
+    # ==== Business fields ====
+    name = fields.Char(string='Number', required=True, readonly=True, copy=False, default='/')
+    date = fields.Date(string='Date', required=True, index=True, readonly=True,
+        states={'draft': [('readonly', False)]},
+        default=fields.Date.context_today)
+    ref = fields.Char(string='Reference', copy=False, readonly=True,
+        states={'draft': [('readonly', False)]})
+    narration = fields.Text(string='Internal Note')
+    state = fields.Selection(selection=[
+            ('draft', 'Unposted'),
+            ('posted', 'Posted'),
+            ('cancel', 'cancelled')
+        ], string='Status', required=True, readonly=True, copy=False, tracking=True,
+        default='draft')
+    type = fields.Selection(selection=[
+            ('entry', 'Journal Entry'),
+            ('out_invoice', 'Customer Invoice'),
+            ('out_refund', 'Customer Credit Note'),
+            ('in_invoice', 'Vendor Bill'),
+            ('in_refund', 'Vendor Credit Note'),
+            ('out_receipt', 'Sales Receipt'),
+            ('in_receipt', 'Purchase Receipt'),
+        ], String='Type', required=True, store=True, index=True, readonly=True, tracking=True,
+        default="entry")
+    user_id = fields.Many2one('res.users', readonly=True, copy=False, tracking=True,
+        states={'draft': [('readonly', False)]},
+        string='Salesperson',
+        default=lambda self: self.env.user)
+    journal_id = fields.Many2one('account.journal', string='Journal', required=True, readonly=True,
+        states={'draft': [('readonly', False)]},
+        domain=lambda self: [('company_id', '=', self.env.company.id)],
+        default=_get_default_journal)
+    company_id = fields.Many2one(string='Company', store=True, readonly=True,
+        related='journal_id.company_id')
+    company_partner_id = fields.Many2one(string='Company Partner', readonly=True,
+        related='company_id.partner_id')
+    company_currency_id = fields.Many2one(string='Company Currency', readonly=True,
+        related='journal_id.company_id.currency_id')
+    currency_id = fields.Many2one('res.currency', store=True, readonly=True, tracking=True, required=True,
+        states={'draft': [('readonly', False)]},
+        string='Currency',
+        default=_get_default_currency)
+    foreign_currency_id = fields.Many2one('res.currency', string='Foreign Currency', # do it in _default_currency_id of account.move.line
+        compute='_compute_foreign_currency_id',
+        help="Technical field used to set the default currency on journal items as their must have a currency_id set only when dealing with foreign currencies.")
+    line_ids = fields.One2many('account.move.line', 'move_id', string='Journal Items', copy=True, readonly=True,
+        states={'draft': [('readonly', False)]})
```

Comment:
all the bellow fields (except `to_check`) are meant to be located in the _invoice fields_ section, isn't it?

Reply:
to check with @william-andre if `to_check` is used only for invoices but I don't think its the case.

Reply:
@smetl all the bellow fields (**except** to_check)

Reply:
To much :sun_with_face: in my office, sorry for the miss-reading

Reply:
@qdp-odoo I think these fields are also relevant for a journal entry and some of them are already used (e.g. `partner_id` or `amount_total`).